interface/models.py

```diff
@@ -152,7 +154,7 @@ class Submission(models.Model):
     history = HistoricalRecords()
 
     def update_state(self):
-        if self.state == self.STATE_DONE or self.evaluator_job_id is None:
+        if self.state in [self.STATE_DONE, self.STATE_ERROR] or self.evaluator_job_id is None :
             return
 
         state = SubmissionScheduler.evaluator.update(self)
```

Comment:
Q: @Jokeswar should we also update on line 164?
If we have
```if state in {Submission.STATE_DONE, Submission.STATE_ERROR}```

Reply:
You are right. We will get our semaphore locked otherwise.

Reply:
Done it @gmuraru and @Jokeswar 👍